pkg/router/template/router.go

```diff
@@ -172,12 +173,20 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	if len(alias.PreferPort) == 0 {
+		for i := range svc.EndpointTable {
+			endpoint := svc.EndpointTable[i]
+			s := endpoint.ID+endpoint.TargetName+endpoint.PortName
```

Comment:
Please put spaces around the +s to be consistent with the rest of the code.

Also, please document what we are hashing in a comment, and why we are adding the name (as protection against rainbow tables).